subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -257,13 +260,15 @@ public DefaultProject(
             services.get(InstantiatorFactory.class).decorateLenient(services)
         );
 
-
         @Nullable DynamicObject parentInherited = services.get(CrossProjectModelAccess.class).parentProjectDynamicInheritedScope(this);
         if (parentInherited != null) {
             extensibleDynamicObject.setParent(parentInherited);
         }
         extensibleDynamicObject.addObject(taskContainer.getTasksAsDynamicObject(), ExtensibleDynamicObject.Location.AfterConvention);
 
+        DynamicObject softwareFeaturesDynamicObject = new SoftwareFeaturesDynamicObject(getSoftwareTypeRegistry(), getSoftwareFeatureApplicator(), this);
```

Comment:
💭 Should the `SoftwareFeaturesDynamicObject` itself be the project-scoped service?  There there would be only one new getter to inject into `DefaultProject`, which would be constructed with the appropriate project instance in the service registry.